### PR TITLE
icu: quell deprecation warnings during build

### PR DESCRIPTION
**Pull request by srl295**
Ref: #950

### CODE REVIEW

**Comment by neheb** [comment]
need to add version bump in releases.json

**Comment by srl295** [comment]
@neheb  let me see if i can make any other improvements then

**Comment by srl295** [comment]
@neheb hm, that requires a new tag. won't this need to be merged first and then tagged? anyway i added a commit with a new version tag

**Comment by neheb** [comment]
it gets tagged after merge.

File: subprojects/packagefiles/icu/source/i18n/meson.build [hunk 1]
```diff
@@ -1,255 +1,5 @@
...
-  #  'decimalformatpattern.cpp',
-  #  'ucurr.cpp',
-  #  'digitlst.cpp',
-  #  'uspoof_wsconf.cpp',
-  #  'decfmtst.cpp',
-  #  'locdspnm.cpp',
-  #  'identifier_info.cpp',
-  #  'filteredbrk.cpp',
-)
+fs = import('fs')
```
Comment (neheb) [inline subprojects/packagefiles/icu/source/i18n/meson.build:1]: the fs variable leaks from the main meson.build file. no need to specify it again.

**Comment by srl295** [comment]
@eli-schwartz  attributed you on the sources.txt commit^

**Comment by srl295** [comment]
this quells almost all of the warnings. There are some about sprintf (which was changed to snprintf or similar in ICU 73)

**Comment by eli-schwartz** [comment]
Nice!

Were there any other changes you wanted to make right now or do you think this is ready to merge now and come back to any other improvements another day?

**Comment by srl295** [comment]
@eli-schwartz I need to do the actual integration which won't be able to depend on this yet due to meson version 😆 

I'll probably need to come back to the switches and stuff later, so maybe let's let this merge and i'll do that another time?

Also I did send out a query to ICU on upstreaming everything.  if the build files were upstreamed, we've already seen good PRs from this project to upstream patches, so I'd assume we could still get help keeping it up to date upstream? (I.e. if the icu/meson.build files were upstreamed, (and i'm not promising that) would they be likely to be maintained there)?

**Comment by eli-schwartz** [comment]
Sure, I am generally happy to be pinged on any issues that a project has with handling tasks in meson. In fact, there is a decent chance I'd spontaneously show up at projects to offer advice, and sometimes to diagnose something as a meson bug and fix it...

I can officially offer to be a code owner and be looped in for reviews too, if that helps.

**Pull request merged**